### PR TITLE
Tiled: added rootPath string so that you can properly parse complex T…

### PR DESCRIPTION
**Pull request by larsiusprime**
…iled maps, added csv parsing to TiledTileLayer.get_tileArray()

### CODE REVIEW

**Comment by larsiusprime** [comment]
Context: an artist delivered a Tiled map to me that I couldn't properly parse with the existing version because it relied on internal file references, which seems like a complex use case that the base parser doesn't seem to account for. This makes it so that it can properly resolve those internal file references.

Also, I added the csv parsing to get_tileArray(), because I don't think it will always be clear to the user what format tiled is representing the internal tile arrays in, and thus know they should use FlxTIlemaps "loadFromCSV" instead -- this causes it to just give you your tile array regardless of what encoding it's using. This happened to me as get_tileArray() just returned nothing, and it wasn't until I went poking around in the file format that I discovered it was because the encoding was "csv" and that wasn't implemented in get_tileArray().

File: flixel/addons/editors/tiled/TiledTileSet.hx [hunk 1]
```diff
@@ -30,7 +31,7 @@ class TiledTileSet
 	
 	public var tileImagesSources:Array<TiledImageTile>;
 	
-	public function new(data:Dynamic)
+	public function new(data:Dynamic, rootPath:String="")
```
Comment (Gama11) [inline flixel/addons/editors/tiled/TiledTileSet.hx]: Hm, we should really use one of the `OneOf` types to clearly define the acceptable types here like we do for `FlxGraphicAsset` etc..

**Comment by Gama11** [comment]
Got a file to test these changes with?

**Comment by larsiusprime** [comment]
@Gama11 just sent you on slack

File: flixel/addons/editors/tiled/TiledTileSet.hx (hunk 1 above)
Comment (larsiusprime) [inline flixel/addons/editors/tiled/TiledTileSet.hx]: Do you want me to go ahead and do that then?

Comment (Gama11) [inline flixel/addons/editors/tiled/TiledTileSet.hx]: Feel free to.

**Comment by larsiusprime** [comment]
@Gama11: Okay, added it. There was already a OneOfTwo with the name "FlxTiledAsset" that didn't seem like the right thing here, so I gave it a clearer name... would that technically be a breaking change? I can revert that.

Also I put in a slightly cleaner way to deal with the legacy Byte classes, but your call on that one too.

File: flixel/addons/editors/tiled/TiledMap.hx [hunk 2]
```diff
@@ -45,14 +45,19 @@ class TiledMap
 	private var noLoadHash:Map<String, Bool> = new Map<String, Bool>();
 	private var layerMap:Map<String, TiledLayer> = new Map<String, TiledLayer>();
 	
+	private var rootPath:String="";
+	
 	/**
 	 * @param data Either a string or XML object containing the Tiled map data
+	 * @param homePath Path to use as root to resolve any internal file references
```
Comment (Gama11) [inline flixel/addons/editors/tiled/TiledMap.hx]: `homePath` in the javadoc here seems incorrect?

File: flixel/addons/editors/tiled/TiledTileSet.hx [hunk 3]
```diff
@@ -200,3 +211,6 @@ class TiledTileSet
 		return new Rectangle((ID % numCols) * tileWidth, (ID / numCols) * tileHeight);
 	}
 }
+
+typedef ValidByteArray = #if (lime_legacy || openfl <= "3.4.0") ByteArray #else ByteArrayData #end;
```
Comment (Gama11) [inline flixel/addons/editors/tiled/TiledTileSet.hx]: Should be `private`.

**Comment by larsiusprime** [comment]
@Gama11  anything else?

**Pull request merged**